### PR TITLE
feat(kubernetes): experimental: cluster, plans & versions 

### PR DESCRIPTION
**Pull request by villevsv-upcloud**

### CODE REVIEW

File: upcloud/kubernetes.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+type KubernetesMaintenancePeriod struct {
+	DOW  string `json:"dow,omitempty"`
+	Hour *int   `json:"hour,omitempty"`
+}
+type KubernetesPlan struct {
+	Name        string `json:"name"`
+	Description string `json:"description"`
+}
+
+type KubernetesError struct {
```
Comment (peknur) [inline upcloud/kubernetes.go:51]: Is this type needed or can we use `upcloud.Error` ?

File: upcloud/request/kubernetes.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/UpCloudLtd/upcloud-go-api/v4/upcloud"
+)
+
+const (
+	KubernetesClusterBasePath string = "/kubernetes"
```
Comment (peknur) [inline upcloud/request/kubernetes.go:11]: Does this need to be exported?

File: upcloud/service/kubernetes.go [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+	"github.com/UpCloudLtd/upcloud-go-api/v4/upcloud/request"
+)
+
+type Kubernetes interface {
+	CreateKubernetesCluster(r *request.CreateKubernetesClusterRequest) (*upcloud.KubernetesCluster, error)
+	DeleteKubernetesCluster(r *request.DeleteKubernetesClusterRequest) error
+	GetKubernetesCluster(r *request.GetKubernetesClusterRequest) (*upcloud.KubernetesCluster, error)
+	GetKubernetesClusters(r *request.GetKubernetesClustersRequest) ([]upcloud.KubernetesCluster, error)
+	GetKubernetesClustersWithFilters(r *request.GetKubernetesClustersWithFiltersRequest) ([]upcloud.KubernetesCluster, error)
+	GetKubernetesKubeconfig(r *request.GetKubernetesKubeconfigRequest) (*string, error)
```
Comment (peknur) [inline upcloud/service/kubernetes.go:17]: Maybe return just string instead of pointer ?

File: upcloud/kubernetes.go (hunk 1 above)
Comment (villevsv-upcloud) [reply upcloud/kubernetes.go:51]: In the API spec there is field `error_details` (string) in the error schema, so created a new type for keeping things a bit cleaner. I can embed the `upcloud.Error`, but let's see if the details is needed in the end.

File: upcloud/request/kubernetes.go (hunk 2 above)
Comment (villevsv-upcloud) [reply upcloud/request/kubernetes.go]: Nope! Fixed.

File: upcloud/service/kubernetes.go (hunk 3 above)
Comment (villevsv-upcloud) [reply upcloud/service/kubernetes.go]: Fixed.

**Comment by villevsv-upcloud** [comment]
LGTM

**Pull request merged**